### PR TITLE
provide a comment for head trackable interface

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: common/types/head_tracker.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ type HeadTracker[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
 //
 //go:generate mockery --quiet --name HeadTrackable --output ./mocks/ --case=underscore
 type HeadTrackable[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
+	// OnNewLongestChain sends a new head when it becomes available. Subscribers can recursively trace the parent
+	// of the head to EVM_FINALITY_DEPTH blocks back. If this is not possible (e.g. due to recent boot, backfill not complete
```
Comment (jmank88) [inline common/types/head_tracker.go:28]: Can we update this env var to the toml config path? Or be even less specific?

Comment (FelixFan1992) [reply common/types/head_tracker.go]: done

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

File: common/types/head_tracker.go [hunk 2]
```diff
@@ -24,6 +24,10 @@ type HeadTracker[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
 //
 //go:generate mockery --quiet --name HeadTrackable --output ./mocks/ --case=underscore
 type HeadTrackable[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
+	// OnNewLongestChain sends a new head when it becomes available. Subscribers can recursively trace the parent
+	// of the head to the EVM finality depth back. If this is not possible (e.g. due to recent boot, backfill not complete
```
Comment (ExtremelySunnyYK) [inline common/types/head_tracker.go:28]: since this would be generalised, maybe we could replace the term `EVM`? or remove it?

**Review by ExtremelySunnyYK — CHANGES_REQUESTED** [review]

File: common/types/head_tracker.go (hunk 2 above)
Comment (FelixFan1992) [reply common/types/head_tracker.go]: done

**Review by jmank88 — APPROVED** [review]

**Review by aalu1418 — APPROVED** [review]

**Review by ExtremelySunnyYK — APPROVED** [review]

**Pull request merged**